backend/src/main/java/com/greenfoxacademy/backend/controller/UserController.java

```diff
@@ -76,4 +77,25 @@ public ResponseEntity<ProfileUpdateResponseDto> userProfileUpdate(
       return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
     }
   }
+
+  /**
+   * This method delete a user from the database.
+   *
+   * @param principal the user that is authenticated
+   * @return a response entity with the status code
+   */
+  @DeleteMapping("/delete-profile")
+  public ResponseEntity<?> deleteProfile(Principal principal) {
+    try {
+      if (principal == null) {
+        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("User is not authenticated");
+      }
+      String username = principal.getName();
+      userService.deleteProfile(username);
+      return ResponseEntity.ok("User deleted successfully");
+    } catch (Exception e) {
+      return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("Unable to delete profile");
+    }
```

Comment:
Can we move the errors or exceptions to the global exception handler to make this thin?

Reply:
@markkovari I made the change on https://github.com/gfa-cc-after/Pet-Clinic-GilmoreDevs/pull/75/commits/0e8326b4e7e532d1744b1bb4f6491edf7aff6bf7 Can you please check if it's good? And if so, resolve this conversation?

Reply:
Looking goof, sorry I forgot to send the answer.